hackathon_site/dashboard/frontend/src/components/orders/OrdersTable/OrdersTable.tsx

```diff
@@ -0,0 +1,164 @@
+import React from "react";
+import {
+    DataGrid,
+    GridColDef,
+    GridRowsProp,
+    GridRenderCellParams,
+    GridEventListener,
+} from "@mui/x-data-grid";
+import { ItemsInOrder, Order } from "api/types";
+import { format, parseISO } from "date-fns"; // to parse date
+import styles from "./OrdersTable.module.scss";
+
+// status icons
+import SubmittedIcon from "../../../assets/images/icons/statusIcons/unfulfilled-status.svg";
+import ReadyForPickupIcon from "../../../assets/images/icons/statusIcons/readyforpickup-status.svg";
+import PickedUpIcon from "../../../assets/images/icons/statusIcons/checkout-status.svg";
+import CancelledIcon from "../../../assets/images/icons/statusIcons/cancelled-status.svg";
+import ReturnedIcon from "../../../assets/images/icons/statusIcons/checkout-status.svg";
+import PendingIcon from "../../../assets/images/icons/statusIcons/pending-status.svg";
+import InProgressIcon from "../../../assets/images/icons/statusIcons/inprogress-status.svg";
+
+interface OrdersTableProps {
+    ordersData: Order[];
+}
+
+interface OrderStateIcon {
+    status: string;
+}
+
+const OrderStateIcon = ({ status }: OrderStateIcon) => {
+    const filterState = status.replace(/\s+/g, "");
+    console.log(filterState);
+
+    return (
+        <div className={styles.container}>
+            <div className={`${styles.container} order-${filterState}-icon`}>
+                {/* <img src={filterState + "Icon"} />
+                <img src={`../../../assets/images/icons/statusIcons/${filterState}-status.svg`} /> */}
```

Comment:
i tried doing something like this on lines 37 and 38 to reduce the lines of code for the icons that appear in the status bubble, but it didn't seem to work... are there any other ways to do this to prevent all the spaghetti below?

Reply:
You could try making a helper function that does it like this, and move the function in `helpers.ts` or another file so it doesn't clutter the file
```javascript
const getStatusIcon = (status) => {
  let iconSrc = InProgressIcon;

  if (status === "Submitted") {
    iconSrc = SubmittedIcon;
  } else if (status === "Ready for Pickup") {
    iconSrc = ReadyForPickupIcon;
  } else if (status === "Picked Up") {
    iconSrc = PickedUpIcon;
  } else if (status === "Cancelled") {
    iconSrc = CancelledIcon;
  } else if (status === "Returned") {
    iconSrc = ReturnedIcon;
  } else if (status === "Pending") {
    iconSrc = PendingIcon;
  }

  return <img src={iconSrc} className={styles.statusIcon} alt={`${status} icon`} />;
};

// Usage:
{getStatusIcon(status)}
```

Reply:
To make it even cleaner(ChatGPT) in typescript:
```typescript
const statusIconMap: { [key: string]: JSX.Element } = {
  Submitted: SubmittedIcon,
  "Ready for Pickup": ReadyForPickupIcon,
  "Picked Up": PickedUpIcon,
  Cancelled: CancelledIcon,
  Returned: ReturnedIcon,
  Pending: PendingIcon,
};

const getStatusIcon = (status: string): JSX.Element => {
  const iconSrc = statusIconMap[status] || InProgressIcon;

  return <img src={iconSrc} className={styles.statusIcon} alt={`${status} icon`} />;
};

```